tools/c7n_oci/c7n_oci/mu.py

```diff
@@ -0,0 +1,620 @@
+import json
+import logging
+import os
+import re
+from abc import ABC, abstractmethod
+
+import oci
+import yaml
+from c7n_oci.constants import COMPARTMENT_IDS
+from oci.events.models import ActionDetailsList, CreateFaaSActionDetails, CreateRuleDetails
+
+from c7n.utils import local_session
+from c7n.version import version
+
+log = logging.getLogger('custodian.oci.serverless')
+
+
+def is_resource_exists(resource):
+    return resource and resource[0].freeform_tags.get('SourceTool') == 'Custodian'
+
+
+class FunctionImageHandler:
+    def __init__(self, session, tenancy):
+        self.session = session
+        self.tenancy = tenancy
+
+    def initialize(self):
+        self.artificats_client = self.session.client('oci.artifacts.ArtifactsClient')
+        self.storage_client = self.session.client('oci.object_storage.ObjectStorageClient')
+        self.identity_client = self.session.client('oci.identity.IdentityClient')
+
+        # Importing here to avoid docker dependency at oracle function runtime
+        import docker
+
+        self.docker_client = docker.from_env()
+        self.repo_name = "cloudcustodian/c7n-oci"
+        self.repo_namespace = self._get_namespace()
+        self.oci_repo = f"{self._get_region_key()}.ocir.io"
+        self.ocir_repo_path = f"{self.oci_repo}/{self.repo_namespace}/{self.repo_name}"
```

Comment:
this is copying from a global read namespace on docker hub, to a private writable one, but the repo name here implies its a shared write space which doesn't exist? ie. how would two users on two different tenants have the same registry name, is that allowed in oci? most CSP registries are against a global namespace by region

Reply:
Each tenancy has a unique tenancy namespace, and the registry name should be unique within that namespace. Additional details can be found at https://docs.oracle.com/en-us/iaas/Content/Registry/Concepts/registryconcepts.htm